app/models/record.server.ts

```diff
@@ -1,59 +1,77 @@
 import { RecordStatus } from '@prisma/client';
-import { prisma } from '~/db.server';
 import dayjs from 'dayjs';
+import { prisma } from '~/db.server';
 
 import type { Record } from '@prisma/client';
 export type { Record } from '@prisma/client';
 
-export async function getRecordsByUsername(username: Record['username']) {
+export interface RecordCreation {
```

Comment:
These two types are clunky.  A type should describe a thing vs an action.

Instead of these names, can you:

1. Use inline types in this file, like `Pick<Record, 'username' | 'type' | 'name' | 'value'>`
2. Instead of using the exported type in other files, inline the objects you pass (I have a note below showing you what I mean)